static/src/javascripts-legacy/projects/common/modules/tailor/tailor.js

```diff
@@ -1,82 +1,26 @@
 define([
-    'Promise',
-    'lib/fetch-json',
-    'lib/cookies',
-    'lib/storage',
-    'lib/report-error',
-    'lib/config'
+    'common/modules/tailor/fetch-data'
 ], function(
-    Promise,
-    fetchJson,
-    cookies,
-    storage,
-    reportError,
-    config
+    fetchData
 ) {
 
-    var URLS = {
-        suggestions: 'https://tailor.guardianapis.com/suggestions?browserId='
-    };
-
-    var browserId = cookies.getCookie('bwid');
-
-    function getURL(type, queryParams) {
-        var baseURL = URLS[type];
-
-        if (!browserId || !baseURL) return;
-
-        baseURL += browserId;
-
-        if (queryParams) {
-            Object.keys(queryParams).forEach(function (key) {
-                baseURL += '&' + key + '=' + queryParams[key];
-            });
-        }
-
-        return baseURL;
-    }
-
     /**
-     * type (required) is a string which should match a key in the URLS object
-     * bypassStorage a boolean, if true don't retrieve data from local storage
-     * queryParams an object literal, each key/value will be query string parameter
-     * eg. {foo:'bar', hello:'world'} translates to ?foo=bar&hello=world
+     * Given a response from tailor, we see if the response has a survey suggestion, and if so return the first
+     * survey suggestion (there should only ever be one, but just in case).
      *
-    **/
-    function fetchData(type, bypassStorage, queryParams) {
-        var url = getURL(type, queryParams);
-
-        return new Promise(function(resolve, reject) {
-            // if no valid url end point, or tailor switch is off, reject
-            if (!url || !config.switches.useTailorEndpoints) return reject();
-
-            var tailorData = bypassStorage ? null : storage.local.get('gu.tailor');
-
-            // if data in local storage return this
-            if (tailorData && tailorData[url]) {
-                return resolve(tailorData[url]);
+     * @returns {Promise.<Boolean>}
+     */
+    function getSuggestion(queryParams) {
```

Comment:
Maybe we should we call this something like' getSuggestedSurvey' as it contains logic to filter out suggestions that aren't surveys?